### PR TITLE
ENH py3 readiness etc

### LINKED ISSUE #154: Support python3
https://docs.python.org/3/howto/pyporting.html seems to be a good source of up-to-date HOWTO.  May be we could even simply switch to python3 altogether while leaving possibility of 3to2 backporting (if such thing is possible) to exist happily on older systems?

### PR DESCRIPTION
**Pull request by yarikoptic**
Having heard Jason's complaints about tests etc decided to swallow the pill and look into "upgrading" code for python3 compatibility (via six).  In my experience with duecredit on one end and fail2ban on the other, I think we better do it sooner than later.

Meanwhile I will do some minimal PEP8 in the code places I touch

Closes #154

### CODE REVIEW

**Comment by yarikoptic** [comment]
OK -- this one is ready for review!  there is only few screw ups:
- didn't figure out why test for our exception handler fails -- so it is skipped ATM
- gitpython has bug in python3 -- filed a report

unrelated -- buildbot tests fail primarily of some screwy DNS behaviour today -- I have setup dnsmasq to proxy/cache it but seems to be either not effective or ...

also there is some level of pain with demanding six 1.9.0 on older releases so I will relax that (view* things were added) so we could just demand six >= 1.8.0

Otherwise -- please review @datalad/developers -- since this will provide good overview of changes you would need to have whenever now we will try to keep code both python2 and python3 compatible.

Cheers!

**Comment by yarikoptic** [comment]
ok -- it is all green!  If no objections -- will merge tomorrow to continue with more interesting stuff ;)
Note that there is now tox.ini -- so you could run sweeps of tests across pythons locally with 'tox' command now.

**Comment by yarikoptic** [comment]
Issue with GitPython was mitigated -- we are pretty much all set for the merge.
Any comments/reservations @datalad/developers ?

**Comment by yarikoptic** [comment]
ok -- since no objections were expressed -- it is getting merged.  Should only get better from here on

**Comment by jgors** [comment]
why not use the new style string formatting (```.format()```)?  it's been around since 2.6 and other than the benefits of doing so, it's def the suggested way in python3?

**Comment by yarikoptic** [comment]
because it is not required for python3 compatibility ;-)  '%' formatting is supported as well and I haven't heard any rumors about its deprecation. 

If someone eventually would do RF for consistency throughout the code-base into `.format` , I would not mind much ;-)  But we better wait for major PRs (pretty much all current ones) to be merged in first before attempting that.

File: datalad/customremotes/base.py [hunk 1]
```diff
@@ -215,14 +217,14 @@ def send(self, *args):
         *args: list of strings
            arguments to be joined by a space and passed to git-annex
         """
-        msg = " ".join(map(str, args)).encode()
+        msg = " ".join(map(str, args))
         if not self._in_the_loop:
             lgr.debug("We are not yet in the loop, thus should not send to annex"
-                     " anything.  Got: %s" % msg)
+                     " anything.  Got: %s" % msg.encode())
```
Comment (jgors) [inline datalad/customremotes/base.py]: jason will make it an argument to debug instead of always interpolating

**Pull request merged**